internal/mesh/internal/controllers/sidecarproxy/mapper/service_endpoints_mapper.go

```diff
@@ -0,0 +1,76 @@
+package mapper
+
+import (
+	"context"
+
+	"github.com/hashicorp/consul/internal/catalog"
+	"github.com/hashicorp/consul/internal/controller"
+	"github.com/hashicorp/consul/internal/mesh/internal/controllers/sidecarproxy/cache"
+	"github.com/hashicorp/consul/internal/mesh/internal/types"
+	"github.com/hashicorp/consul/internal/resource"
+	pbcatalog "github.com/hashicorp/consul/proto-public/pbcatalog/v1alpha1"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+type Mapper struct {
+	cache *cache.Cache
+}
+
+func New(c *cache.Cache) *Mapper {
+	return &Mapper{
+		cache: c,
+	}
+}
+
+// MapServiceEndpointsToProxyStateTemplate maps catalog.ServiceEndpoints objects to the IDs of
+// ProxyStateTemplate.
+// For a destination proxy, we only need to generate requests from workloads this "endpoints" points to
+// so that we can re-generate proxy state for the sidecar proxy.
+// If this service endpoints is a source for some proxies, we need to generate requests for those proxies as well.
+// so we need to have a map from service endpoints to source proxy Ids.
```

Comment:
I think these two comments about the `service->workload->template` and `service->upstream->template` would work better as clarifiers if they were placed just ahead of the portion of the method that does that distinct job.

Reply:
done!